### DISCUSSION TITLE
solved: Could not load the Qt platform plugin "xcb"

### DISCUSSION THREAD

**Discussion by ljubomirb** [NVlabs/instant-ngp#300] [Q&A] [answered]
I will leave here solution for a problem that I could not fix for days, hope it would help someone:
So, the error was:
```
QObject::moveToThread: Current thread (...) is not the object's thread (...).
Cannot move to target thread (...)
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/(...)site-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

```
I was trying hard, but could not fix it. I was downloading literally hundreds of different packages, over and over again, trying all sorts of "hacks" found online, eventually trying anaconda, a massive download...
In the end, I don't know if that was the fix, because my system is about to be reinstalled after all of my "trying", BUT, after I wrote this in console:
`export QT_QPA_PLATFORM=offscreen `
everything worked.
Again, don't know if this was solution OR any of hundreds of packages installed, but try this first. If it works, maybe add this to Q&A .
Good luck!

**Comment by Tom94** [comment]
Hi there, thanks for reporting. This seems to have been a problem with the `cv2` Python package and how it loops in Qt -- the __instant-ngp__ codebase itself doesn't actually use Qt anywhere. Cheers!

**Comment by Ednaordinary** [comment]
Not sure exactly why, but this made it work for me! original error was

`qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/usr/local/lib/python3.9/dist-packages/opencv_python-4.5.5.64-py3.9-linux-x86_64.egg/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc.`

I agree this should be added to the Q&A, even if its a dependency and not instant-ngp that is causing trouble

**Comment by ben-milanko** [comment, marked as answer]
I ran into this issue, it seems to be from open-cv https://github.com/opencv/opencv-python/issues/46.


Using open-cv-python-headless instead seems to work for me: https://stackoverflow.com/a/72090539

```
pip uninstall opencv-python
pip install opencv-python-headless
```

**Reply by Jack1412** [reply]
It works, thanks!

**Reply by shivamkumarpanda** [reply]
Worked. Thanks

**Reply by bouboulov** [reply]
Thanks !

**Reply by flanremi** [reply]
help me a lot ,  Thanksssssssss!

**Reply by benz725** [reply]
It really works!  Thank You!!!

**Reply by Think-man-blog** [reply]
It really works for me! Thank you a lot!!!

**Reply by mr-loop-1** [reply]
Thanks a lot

**Reply by Riponcs** [reply]
Worked for me.

**Reply by AI-Tianlong** [reply]
Nice!!!!!!! Right suggestion!
Thanks!!!

**Reply by Special256** [reply]
It works

**Reply by 35grain** [reply]
Big W Ben

**Reply by JYS997760473** [reply]
cool it works

**Reply by bitshifter52** [reply]
I uninstalled and reinstalled opencv-python-headless and that worked. Thank you for the tip.

**Reply by dylanosaur** [reply]
hmm didnt work for me

**Reply by dylanelliotttamu** [reply]
did not work for me either python=3.8.12 and conda

**Reply by gelodefaultbrain** [reply]
Works well thanks!

**Reply by rvbabera** [reply]
Worked for me.

**Reply by sayco** [reply]
worked for me, thanks

**Reply by Benjizhang** [reply]
worked for me, thanks a lot

**Reply by NimaDolatabadi** [reply]
opencv-contrib-python> I ran into this issue, it seems to be from open-cv [opencv/opencv-python#46](https://github.com/opencv/opencv-python/issues/46).
> 
> Using open-cv-python-headless instead seems to work for me: https://stackoverflow.com/a/72090539
> 
> ```
> pip uninstall opencv-python
> pip install opencv-python-headless
> ```

this did not work for me at all.
i have ubuntu 20.04
anaconda with env configured by python 3.10
opencv-python && opencv-contrib-python
opencv-python-headless

none of the workd.
because opencv-python-headless does not support cv2.imshow()
i have to use alternative like plt and by using that i cannot use cv2.waitkey(0) cause there is no alternative.
any idea? been searching about this for 1 month and still got nothing.

**Reply by fathelmarc** [reply]
thank you so much, it indeed work

**Reply by Efoblack** [reply]
works for me. thanks a lot

**Reply by Mozhgan91** [reply]
ubuntu 20.04
worked for me. thanks a lot

**Reply by sowji-lekha13** [reply]
Thank you so much!!

**Reply by xmkkkkkk** [reply]
It works! Thanks you!

**Reply by Borzygniew** [reply]
Linux 5.15
working, thanks! 😎

**Reply by vasuprint** [reply]
Thank you so much

**Reply by gregorarmstrong5** [reply]
Thanks so much! Works on Linux 5.15!

**Reply by Jayesh6999** [reply]
Thanks a lot! It worked out for me.

**Reply by Pandafishie** [reply]
This topic still seems to be an issue a year later. Thanks for your solution! Worked wonders for me.

**Reply by zzy-ucas** [reply]
Thanks a lot! It works for me.

**Reply by tabarkarajab** [reply]
this worked

**Reply by leoguen** [reply]
This worked for me thanks!

**Reply by HakanTuncel42** [reply]
thank you very much big man

**Reply by RylonW** [reply]
It works! But when I only changed the cv2 to headless version, I came the problem:
`'numpy._DTypeMeta' object is not subscriptable`
I solved it by degrading the numpy version to 1.20.0. According to : [https://stackoverflow.com/questions/68916893/typeerror-numpy-dtypemeta-object-is-not-subscriptable](https://stackoverflow.com/questions/68916893/typeerror-numpy-dtypemeta-object-is-not-subscriptable)

**Reply by mmmiiissshhhaaa** [reply]
@NimaDolatabadi were you ever able to find a solution? I'm running into the same problem.

**Reply by NimaDolatabadi** [reply]
i am using pycharm i just found out that i must not define costume path for cv2 in pycharm. that will do the job.
regard

**Reply by abnormal749** [reply]
But it now cannot use `cv2.imshow` after install `opencv-python-headless`

```
cv2.error: OpenCV(4.8.1) /io/opencv/modules/highgui/src/window.cpp:1272: error: (-2:Unspecified error) The function is
not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install
libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvShowImage'
```

**Reply by LunaRose12** [reply]
Thank you for taking the time to post that for the idiots out there trying to learn computers as in myself

**Reply by luisjuarez099** [reply]
Thank you for your response. Fortunately, my boss didn't dismiss me. :no_good_man: :eyes:

**Reply by BoZhiStudying233** [reply]
@abnormal749 I met the same problem. Have you solved it? I solved this problem by isolating the two programs via a virtual environment.

**Reply by joyehsu** [reply]
Works for me too, thanks very much~
My system is Win10 + WSL2

**Reply by guliqianxun** [reply]
For me, this causing I use vscode connect to host, and no screen to show. I use realVNC connect the host and run the testing code there is no error then.

**Reply by Aut6ot** [reply]
Thank you! This helped a lot

**Reply by Twograpefruit** [reply]
Thank you very much, the path I reported the error did contain cv2'qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/home/anxing/.local/lib/python3.8/site-packages/cv2/qt/plugins"', and I successfully solved it.

**Reply by Vinson-Tang** [reply]
It works,thanks!

**Reply by RainloongCao** [reply]
Thanks!! It does work

**Reply by mrizkipp** [reply]
Thanks! It works

**Reply by 9luap** [reply]
Thank you so much, it worked for me as well!

**Comment by MashyBasker** [comment]
Hi, were you using a virtual environment for installing python packages? I was facing the exact same problem and 30 mins later I realized my virtual environment was not activated.

**Reply by hoods-py** [reply]
i am using a virtual environment
and i'm having the same issue

**Comment by yojetsharma** [comment]
I am trying to run rstudio in a conda environment on an institutional server and I am getting the following error:
```
(rstudio) [yojetsharma@pakeeza ~]$ rstudio
TTY detected. Printing informational message about logging configuration. Logging configuration loaded from '/etc/rstudio/logging.conf'. Logging to '/home/praghu/yojetsharma/.local/share/rstudio/log/rdesktop.log'.
No protocol specified
Fontconfig warning: ignoring UTF-8: not a valid region tag
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
This plugin does not support propagateSizeHints()
This plugin does not support raise()
Failed to create OpenGL context for format QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile) 
Aborted
[5851:5851:0100/000000.063786:ERROR:zygote_linux.cc(607)] Zygote could not fork: process_type renderer numfds 3 child_pid -1
[5851:5851:0100/000000.063868:ERROR:zygote_linux.cc(639)] write: Broken pipe (32)
```

**Reply by XeJames** [reply]
After trying a whole lot of libxcb installations (all of which did nothing), I did this
`sudo apt install libxcb-* `
No idea WHICH library was the one, but now that I have them all, it's working.

**Reply by laban11** [reply]
This helped me, Thanks

**Reply by marcin-osowski** [reply]
Tried XeJames's idea, found out that for me it was `libxcb-cursor0` that fixed the problem.

**Reply by Borja21091** [reply]
It worked! Thanks!

**Reply by kassbohm** [reply]
Worked also for me on Linux Mint.

**Reply by Kupofty** [reply]
Also worked for me on Ubuntu20 after the last update of QtCreator, thanks a lot !

**Reply by montanaviking** [reply]
running Ubuntu 22.04. Sadly installing opencv-python-headless did not work for me

**Reply by The-R4V3N** [reply]
Thanks this finally worked

**Reply by fattypiggy** [reply]
This one works for me, thanks a lot.

**Comment by israelins85** [comment]
To solve this just run:

sudo apt-get install libxcb-xinerama0

**Reply by mostafa1912** [reply]
worked for me

**Reply by siddharth-57** [reply]
This worked on WSL version 20.4 thanks!

**Reply by shuanghan0922** [reply]
This is working on Ubuntu22.04! Thanks!

**Reply by mayankdatagrid** [reply]
Works for me

**Reply by rafaelpadilla** [reply]
What would be the similar package for Windows?  Any idea? @MathewsJosh

**Reply by ansarker** [reply]
worked

**Reply by karastoyanov** [reply]
Life savior! Worked on ubuntu 22.04 LTS

**Reply by vutriancode** [reply]
worked for me thank you

**Reply by daxiongshu** [reply]
worked for me. thanks!

**Reply by sujonahmed2500** [reply]
Thanks!

**Reply by shaojun** [reply]
worked in windows WSL2

**Reply by AmandineGit** [reply]
worked for me too. thanks!

**Reply by wuyiulin** [reply]
it work for me in Ubuntu 22.04, THX.

**Reply by aino-debasish** [reply]
This seems to be the real solution for Ubuntu. There should be a way in github to highlight this as a solution :)

**Reply by amitpd1** [reply]
worked for ubuntu

**Reply by ArunChamiyappan** [reply]
The same here, Thank you

**Reply by Lijuming33** [reply]
ubuntu 2204，有用！

**Reply by xyf9474** [reply]
牛逼

**Reply by ll553664391** [reply]
works for me . on ubunut 20.04

**Reply by Joan-Hou** [reply]
> To solve this just run:
> 
> sudo apt-get install libxcb-xinerama0

very helpful! worked for me! I am using ubuntu20.04, windows10, wsl1, on conda enviroment.

**Reply by whoisltd** [reply]
work for me ubuntu 24.04

**Reply by flying-unicorn-97** [reply]
doesn't work for me ubuntu 20.04

**Comment by Jalagarto** [comment]
https://stackoverflow.com/a/68058308/9033534  -  first answer

> for pyqt6 in ubuntu 20.04, following spell in forum linked above, solved the problem for me `sudo apt-get install '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev` 
> Author: harish
>  Oct 21, 2021

**Reply by tomnattle** [reply]
3Q its work for me

**Reply by jasonltorchinsky** [reply]
This is what worked for me! Running WSL and with a fresh Conda install.

**Reply by tatyanamincheva** [reply]
Worked like magic on 22.04, thanks!

**Reply by jinzzasol** [reply]
This works for me. WSL with Ubuntu, miniconda.

**Reply by Zhiyu-h** [reply]
this works for me with Ubuntu22.04 WSL2

**Reply by sidb98** [reply]
worked for me on Ubuntu 22.04.2 LTS WSL2

**Reply by mehdirexon** [reply]
worked thanks

**Reply by Malvin555** [reply]
Finally Its Work

**Reply by shenyimings** [reply]
Oh my god, it works for me! Ubuntu 22.04

**Reply by morais1244** [reply]
worked for me thanks!!

**Reply by venkatasubbareddyy** [reply]
Worked for me. Thanks !!

**Reply by Diamaa12** [reply]
Worked for me. Just Thanks

**Reply by wolfyc** [reply]
worked for me too !! thank youuu !

**Reply by Harry-WXH** [reply]
It worked! THX!

**Reply by diegoOPires** [reply]
worked for me, thanks!

**Reply by dashwood99** [reply]
thx a lot,worked for me!!

**Reply by wilson1375** [reply]
Thanks

**Reply by criogenox** [reply]
It works!!!! Thanks, my friend.

**Reply by Drone944** [reply]
Thanks Bro, it works, saved me a lot of time.

**Reply by interval1066** [reply]
Thanks Jalagarto, your apt install chain worked for me as well. I know a lot of AI tools as well as everything else uses python, it seems, but I resent having to install a bunch of junk I don't use just to get qtcreator working. I didn't ask the updater to install this copilot client on my qt stack.. especially anything made with python. I really resent this.

**Reply by mahdi3100** [reply]
thanks mate

**Reply by zezar98** [reply]
It worked for me thanks!  I just installed Debian 12.5 and anaconda-navigator wasn't working for me. After install those packages it worked.

**Reply by gitfourteen** [reply]
> https://stackoverflow.com/a/68058308/9033534 - first answer
> 
> > for pyqt6 in ubuntu 20.04, following spell in forum linked above, solved the problem for me `sudo apt-get install '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev`
> > Author: harish
> > Oct 21, 2021

With win10, using WSL2 ubuntu 20.04 LTS, python 3.9 and matplotlib to plot a scatter, then got the error: `t.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.`  This should be your first try!

**Reply by YuZhang-GitHub** [reply]
Absolute saviour. Only command so far that works for me:
Ubuntu 22.04.3 LTS in WSL version: 2.2.4.0 with Windows 10 host

**Comment by tomnattle** [comment]
sudo apt-get install '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev

this work for me , i use qt in docker

**Reply by tomnattle** [reply]
```
168  pip3 install labelImg
  169  labelImg
  170  export DISPLAY=192.168.56.1:0.0
  171  labelImg
  172  export QT_DEBUG_PLUGINS=1
  173  labelImg
  174  apt-get install libxcb-xinerama0
  175  labelImg
  176  cd /home/brainiac/Qt/Tools/QtCreator/lib/Qt/plugins/platforms/
  177  labelImg
  178  ldd /usr/lib/qt/plugins/platforms/libqxcb.so
  179  pip uninstall opencv-python
  180  pip install opencv-python-headless
  181  labelImg
  182* pip uninstall opencv-python-
  183  pip install opencv-python
  184  sudo apt-get install '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev    
  185  apt-get install '^libxcb.*-dev' libx11-xcb-dev libglu1-mesa-dev libxrender-dev libxi-dev libxkbcommon-dev libxkbcommon-x11-dev
  186  labelImg
  187  history
```

**Reply by darkrams98** [reply]
This worked for me : ubuntu 22.04

**Reply by JoshuaTokisaki** [reply]
This worked for me: ubuntu 22.04 WSL 2

**Reply by jinhoishere** [reply]
This worked for me : Ubuntu WSL 2 22.04.4 LTS

**Reply by zym198098** [reply]
我用wsl2 ubuntu 22.04 有用。多谢

**Reply by Wangchangcong** [reply]
This worked for me : wsl1 ubuntu 20.04

**Reply by pporwal015** [reply]
Thanks. Worked for me on WSL ubuntu

**Comment by bhjolly** [comment]
> export QT_QPA_PLATFORM=offscreen

I ran across this thread trying to diagnose the same issue trying to run Agisoft Metashape on a GPU-equipped-but-headless server. Exporting that environment variable fixed it.

**Reply by satpalsr** [reply]
This helped. Thanks. I am runnning in ec2-instance.

**Reply by maple367** [reply]
This works in WSL2 debian with conda environment. Thanks.

**Reply by BryceWayne** [reply]
This worked Ubuntu+WSL+Conda+Python3.10

**Reply by AmrinKareem** [reply]
This worked for me in Linux with conda environment. Thank you!

**Reply by m1ndcrash** [reply]
Oh my god. 4 hours. This one saved me. PS training nerf

**Reply by J1ungle** [reply]
This works for ubuntu 20.04 in docker!

**Reply by Kklyee** [reply]
wsl2  not work for me , why?

**Reply by sopryshko** [reply]
Thank you! Worked for ubuntu:24.04 in Docker

**Comment by Petros626** [comment]
My solutions I've tested on my code.

If you get Could not load the Qt platform plugin "xcb"after import `cv2`
fix 1: use `os.environ.pop("QT_QPA_PLATFORM_PLUGIN_PATH")` in code
fix 2: try pip install `opencv-python-headless` (without GUI)
fix 3: delete the _libqxcb_ from  "/home/usernamexyz/.local/lib/python3.9/site-packages/cv2/qt/plugins/platforms

**Reply by Daa98** [reply]
fix 3: delete the libqxcb from "/home/usernamexyz/.local/lib/python3.9/site-packages/cv2/qt/plugins/platforms

is useful

Thanks

**Reply by HasarinduPerera** [reply]
Thanks, this is useful indeed!

**Reply by DatleUNL** [reply]
Fix 1 worked for my Kinz library that uses opencv-contrib-python. Thanks!

**Reply by WEIIEW97** [reply]
Fix 1 solved mine. Thanks a lot!

**Reply by yaioqin** [reply]
fix 3 is useful. Thanks

**Reply by wh1t3h47** [reply]
thank you

**Reply by Petros626** [reply]
@Daa98 @HasarinduPerera could you guys help me with something related to KITTI, BEV, coordinate transformation, if you're familiar with it?

**Comment by masuraoru** [comment]
In my case it was conflict in Qt installation on the system with the Qt that is shipped with OpenCV. Solved on Ubuntu 22.04 Desktop by removing opencv-python:
`pip3 uninstall opencv-python`
and reinstaing with --no-binary option:
`pip3 install --no-binary opencv-python opencv-python`
Warning: opencv-python compiling may take some time.

**Reply by AnshuDFKI** [reply]
This is the only solution which worked for me. Thank You

**Reply by xmlyqing00** [reply]
This is the most decent way to solve this problem.

**Reply by abnormal749** [reply]
It's the only solution really helps for my program with qt and `cv.imshow`

**Reply by Lwy-1998** [reply]
This is really help！！！

**Reply by JasOleander** [reply]
You saved my day. This is the only answer for both qt and cv2 errors.
Thank you so much. :)

**Comment by ljjTYJR** [comment]
My problem:
```
qt.qpa.xcb: could not connect to display
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, xcb.
```

I use the command:
```
export DISPLAY=:0
```
works for me...

**Reply by eneserdo** [reply]
Worked for WSL, thanks

**Reply by tymho** [reply]
Worked for me! I think this might be related to SSH?

**Reply by kelogs** [reply]
> SSH

Sure thing, no display through SSH sessions!

**Reply by Soka21** [reply]
This worked for me too, thanks!

**Reply by beizhengren** [reply]
> > SSH
> 
> Sure thing, no display through SSH sessions!
hey bro, have you solved it in ssh?

**Comment by trojanof** [comment]
Thanks for solution! `export QT_QPA_PLATFORM=offscreen` works for me! I had an issue running streamlit app on a server with ubuntu and music21 python module. I had musescore installed there but faced the qt related problem when trying to display music scores. Setting that variable solved my problem.

**Reply by opticalor** [reply]
Thats not work,but i use `export QT_QPA_PLATFORM_PLUGIN_PATH=/path/to/your/qt/directory/plugins`,is run now.(dont forget unset QT_QPA_PLATFORM if you accept before this)

**Comment by zzhuolun** [comment]
I found [this answer ](https://stackoverflow.com/questions/64695574/after-installing-pyqt5-5-14-0-error-shows-as-could-not-load-the-qt-platform-pl)that showed how to narrow down and locate the problem very helpful.

In my case, installing the specific missing libraries AND reinstalling PyQt5 solved the problem.

**Comment by Truthuleave** [comment]
No error appears through `export QT_DEBUG_PLUGINS=1` when in conda venv.
Quitting from conda venv works for me~

**Comment by liangqx-hku** [comment]
> I will leave here solution for a problem that I could not fix for days, hope it would help someone: So, the error was:
> 
> ```
> QObject::moveToThread: Current thread (...) is not the object's thread (...).
> Cannot move to target thread (...)
> qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/(...)site-packages/cv2/qt/plugins" even though it was found.
> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
> ```
> 
> I was trying hard, but could not fix it. I was downloading literally hundreds of different packages, over and over again, trying all sorts of "hacks" found online, eventually trying anaconda, a massive download... In the end, I don't know if that was the fix, because my system is about to be reinstalled after all of my "trying", BUT, after I wrote this in console: `export QT_QPA_PLATFORM=offscreen ` everything worked. Again, don't know if this was solution OR any of hundreds of packages installed, but try this first. If it works, maybe add this to Q&A . Good luck!

This answer just turns off the screen. Maybe this cmd could help you guys fix the issue.   `sudo apt install libxcb-xinerama0`

**Comment by sajadalipour** [comment]
It works for me after installing opencv-contrib-python-headless

**Comment by 1850298154** [comment]
您的邮件已收到！祝工作顺利！

**Comment by eavelardev** [comment]
`export XDG_CURRENT_DESKTOP=Unity` could work for Ubuntu, but I have an undesired behavior where PyQt windows and dialogs aren't centered

**Comment by CrikersPL** [comment]
`flatpak override --user --env=QT_QPA_PLATFORM=wayland org.kde.dolphin`
(org.kde.dolphin) in my case

Worked for me

**Comment by stevenlafl** [comment]
I got this error when loading a QT6 program that didn't ship the required QT6 plugins. In particular, obs under rockchip repository for aarch64. Instead, the plugin package installed was a QT5 one, so essentially it wouldn't find any DDE plugins.

```bash
apt-get install qt6-qpa-plugins
```
will install the `libqxcb.so` file required along with [libqeglfs.so](#), [libqminimalegl.so](#), [libqoffscreen.so](#), [libqvnc.so](#), [libqlinuxfb.so](#), [libqminimal.so](#) and [libqvkkhrdisplay.so](#).

**Comment by AndrewHart1** [comment]
1.7.9
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
QObject::moveToThread: Current thread (0x1f63490) is not the object's thread (0x2de7af0).
Cannot move to target thread (0x1f63490)

qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "/home/smart-daq/anaconda3/lib/python3.9/site-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb, eglfs, minimal, minimalegl, offscreen, vnc, webgl.

Aborted (core dumped)

i am getting this error and after trying all the aformentioned methods it is still not working. does anyone have any ideas

**Comment by Franciscofmv** [comment]
Thanks for posting this! `ENV QT_QPA_PLATFORM=offscreen `  in the Dockerfile solved it for me when building a docker container. I was trying to run PyQt5 in a container but I was having that error. Thanks again! Note: I first needed to do `RUN chmod -R 777 /opt/miniconda3;    yum -y install mesa-libGL`  in the Dockerfile to install the graphic library.

**Comment by Abbsalehi** [comment]
This worked for me.

```bash
sudo apt-get install libx11-xcb1 libxcb1 libxcb-glx0 \
libxcb-keysyms1 libxcb-image0 libxcb-shm0 libxcb-icccm4 \
libxcb-sync1 libxcb-xfixes0 libxcb-shape0 libxcb-randr0 \
libxcb-render-util0 libxcb-render0 libxcb-xinerama0 libxcb-xkb1 libxkbcommon-x11-0
```

**Comment by space-pixel** [comment]
When I tried these solutions on windows 10 while using a container the code compiled but did not output the visual. The fix I found was installing Xming here: https://sourceforge.net/projects/xming/ and using this line: docker run -e DISPLAY=host.docker.internal:0.0 your-qt-app to run the container.

**Comment by johndpope** [comment]
`conda install -c conda-forge qt
`

**Comment by santo4ul** [comment]
sudo apt-get remove libxcb-xinerama0
sudo apt-get install libxcb-xinerama0

This works for me. Hope this helps someone.

**Reply by gayu-thri** [reply]
This helped me, thanks.

**Comment by IgdaliasCabamba** [comment]
`sudo dnf install qt5-qtbase-devel` Worked for me on **Fedora 40** with  **xfce**

**Comment by JLee2022** [comment]
I fixed it by `sudo apt-get remove libxcb-xinerama0
sudo apt-get install libxcb-xinerama0`.
and I set `export QT_QPA_PLATFORM=xcb`.
Notice that i am using vscode to debug python files.

**Reply by JLee2022** [reply]
I found that i need to run the code in terminal. Than I can run it in my vscode. strange...

**Comment by ashrea06** [comment]
![2024-12-01 07_42_59-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/43a43f99-d280-4009-96f0-b7fb2124dbbd)
![2024-12-01 07_43_12-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/e8409e3c-8d54-47a2-9624-ad17fff07455)
![2024-12-01 07_44_14-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/b94b2eea-5930-49d4-867d-78be95ac2237)
![2024-12-01 07_46_10-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/27bf8fc7-28d8-4514-b4b3-8c1d15abdef7)
![2024-12-01 07_52_48-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/a61a0a46-a569-48d2-b390-3e0a596bab7a)
![2024-12-01 07_54_52-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/b3bc941c-0d29-4974-a9df-81ccb607ccb8)
![2024-12-01 07_59_36-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/99e79132-fff7-4e75-a17e-f3f11f8ce2de)
![2024-12-01 08_00_06-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/84ae8136-7499-4e09-a804-839caf4619f3)
![2024-12-01 08_00_30-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/6683b557-0f1a-427c-9973-fc3bfcefe1b9)
![2024-12-01 08_00_57-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/50d69ce0-d56d-434b-b6ae-aaee513a61aa)
![2024-12-01 08_04_34-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/aca03e5e-0216-4d1e-8c4a-35aa2d5aae8f)
![2024-12-01 08_08_30-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/672f6786-3517-4fa2-9f0a-f81c8e56d867)
![2024-12-01 08_11_58-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/52ff4415-f3c3-47da-b2ba-4e5abe889c61)
![2024-12-01 08_12_41-Qt6 Installation on Ubuntu from command line - C++ binding example with QPropert](https://github.com/user-attachments/assets/3d3491ab-e438-4ee4-b200-1722e5660fd9)
![2024-12-01 17_19_59-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/d463e316-910b-4786-9208-e01d09998d51)
![2024-12-01 19_25_17-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/e11087b5-f618-4691-98bb-a09888d262ae)
![2024-12-01 20_07_51-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/0559fce7-82c7-437b-b240-e2aac7317960)
![2024-12-01 20_13_53-Meta AI - Avast Secure Browser](https://github.com/user-attachments/assets/277859ca-48bf-45fa-8a38-8d04b876d8e0)
![2024-12-02 06_06_53-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/40c8d8a7-7319-4647-95a9-32c65a6a5f83)
![2024-12-02 06_10_14-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/b32d7a94-59e7-4fbd-9904-e89d3b50581e)
![2024-12-02 19_32_04-Meta AI - Avast Secure Browser](https://github.com/user-attachments/assets/85b14c8e-f599-4b28-aef0-6955d6d5f918)
![2024-12-02 19_32_26-Meta AI - Avast Secure Browser](https://github.com/user-attachments/assets/399adba1-4004-4a2a-9f92-e8f644616291)
![2024-12-02 20_53_02-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/ced37851-a522-4c2b-90ca-024d53ab5248)
![2024-12-02 20_54_09-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/b09e29fd-5752-4744-9cce-5cea96589146)
![2024-12-02 20_54_30-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/db2eab63-a279-41ec-808a-ee0a1981af3b)
![2024-12-02 20_56_13-Unix   Linux_ Could not load the Qt platform plugin _xcb_ (3 Solutions!!) - YouT](https://github.com/user-attachments/assets/e7e294b1-e85e-494d-8b61-73b4bfccf913)
![2024-12-02 20_59_38-TCMUbuntu 64-bit - VMware Workstation 17 Player (Non-commercial use only)](https://github.com/user-attachments/assets/0c2e1c47-fc69-4700-b45d-131b7e52614f)
![2024-12-02 21_06_08-Meta AI - Avast Secure Browser](https://github.com/user-attachments/assets/67da358a-1d11-4602-94f5-e5026885ffcb)
Hey everyone if i can get some help in regards to these snapshots added, same error message( .  Using [Ubuntu 24.04 LTS] - Genymotion required qt6 +qtcreator before it could run. I tried all the steps from above, however, the error message persists: 

"This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: minimal, xcb.

Aborted (core dumped)


i tried almost all troubleshooting mentioned on this post, and also went through couple of Videos. Let me know if i could provide any further information.

**Comment by osering** [comment]
Under Ubuntu 24.04 (on headless server) this one helped (to pipe through ssh to laptop):
`sudo apt install -y libxcb-cursor-dev`

**Comment by conscious-zombie** [comment]
In case one is working on a Linux system, does not have root administrative privileges and cannot use `sudo` or `apt`, I tried:

> `mamba install -c conda-forge pyqt`
> `mamba install -c conda-forge xcb-util-cursor xorg-libxinerama`

And it worked!
